sqlglot/dialects/duckdb.py

```diff
@@ -89,6 +89,11 @@ def _struct_sql(self: generator.Generator, expression: exp.Struct) -> str:
 def _datatype_sql(self: generator.Generator, expression: exp.DataType) -> str:
     if expression.is_type("array"):
         return f"{self.expressions(expression, flat=True)}[]"
+
+    # Type TIMESTAMP WITH TIME ZONE does not support any modifiers
+    if expression.is_type("timestamptz", "timetz"):
+        return "TIMESTAMPTZ"
```

Comment:
For give my ignorance, but should this be returning `TIMESTAMPTZ` even though `expression` can be either a timestamptz or timetz?

Reply:
DuckDB doesn't support `timetz` as a type - see https://github.com/tobymao/sqlglot/pull/2041#discussion_r1291602767 for more context

Reply:
Are you on an older version?

It works for me:

```
D select cast(now() as time with time zone);
┌────────────────────────────────────┐
│ CAST(now() AS TIME WITH TIME ZONE) │
│        time with time zone         │
├────────────────────────────────────┤
│ 17:14:16.675+00                    │
└────────────────────────────────────┘
```

Reply:
Works in Python too:

```
❯ python -c 'import duckdb; print(duckdb.execute("select cast(now() as time with time zone)").fetchall())'
[(datetime.time(17, 15, 38, 764000),)]
…/ibis on  add-trino-source is 📦 v6.1.0 via 🐍 v3.10.12 via ❄️  impure (shell)
```

Reply:
`timetz` looks valid too:

```
❯ python -c 'import duckdb; print(duckdb.execute("select cast(now() as timetz)").fetchall())'
[(datetime.time(17, 17, 40, 982000),)]
```

Reply:
Oh, whoops, you're actually right. I must've assumed that they didn't support this type because it's not mentioned in their [documentation](https://duckdb.org/docs/sql/data_types/overview.html) (only `timestamp with time zone` is), but I should've definitely double-checked.

Will update, thanks for catching this!

Reply:
Alrighty, fixed!